src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] arrayOfObjects;
+    private int size;
+
+    public ArrayList() {
+        arrayOfObjects = new Object[DEFAULT_SIZE];
+    }
 
     @Override
     public void add(T value) {
+        if (size == arrayOfObjects.length) {
+            growCapacity();
+        }
 
+        arrayOfObjects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound");
+        }
+
+        int arrayLength = arrayOfObjects.length;
+        Object[] tempo = new Object[arrayLength + 1];
```

Comment:
same comment as for remove() method - let's remove the `tempo` variable here and use `System.arraycopy` only once
we can move part of the elements inside the same arrayOfObjects array, and we don't need to increase its size - instead we might do the same check as for `add(T value)` method to be sure that the new element will fit